src/arenaui.cpp

```diff
@@ -264,13 +275,32 @@ bool MouseClickHandler::eventFilter(QObject* obj, QEvent* event)
 //-------------------------------------------------------------------------------
 // Subclassed QGraphicsScene for a BUG [QTBUG-10138]
 // http://www.qtcentre.org/threads/36953-QGraphicsItem-deselected-on-contextMenuEvent
-QArenaScene::QArenaScene(QWidget *parent) : QGraphicsScene(parent){}
+QArenaScene::QArenaScene(QWidget *parent) : QGraphicsScene(parent){
+    this->setItemIndexMethod(QGraphicsScene::NoIndex);
+    connect(this,SIGNAL(selectionChanged()),SLOT(checkSelection()));
+}
 
 void QArenaScene::mousePressEvent(QGraphicsSceneMouseEvent *event)
 {
     if (event->button() == Qt::RightButton) event->accept();
     else QGraphicsScene::mousePressEvent(event);
 }
+
+void QArenaScene::checkSelection()
+{
+    QList<QGraphicsItem*> tempList = this->selectedItems();
+    if(tempList.size()>1) selectionTreeWidget->setHidden(false);
+    else selectionTreeWidget->setHidden(true);
+
+    int color = 14;
+
+    foreach(QGraphicsItem* item, tempList){
+        if(item->childItems().size()){
+            ((QCasuSceneGroup*)item)->setGroupColor((Qt::GlobalColor) color);
+            ((QCasuSceneGroup*)item)->treeItem->setTextColor(0,(Qt::GlobalColor) color++);
+        }
+    }
+}
```

Comment:
When group is selected, it's groupTree item is shown in same color as that group on graphicsScene. Also casuSceneItem is now colored in groupColor if selected for easier visual separating and group affiliation